networkx/linalg/algebraicconnectivity.py

```diff
@@ -244,7 +244,7 @@ def P(x):
             W -= (W.T * X * X.T).T
             project(W)
             # Compute the diagonal of P * L * P as a Jacobi preconditioner.
-            D = L.diagonal()
+            D = L.diagonal().astype(float)
             D += 2. * (asarray(X) * asarray(W)).sum(axis=1)
```

Comment:
Honestly I didn't really try to understand the code, but from the error given by TravisCI, NumPy is just refusing to add up two array of different types, so even if combining these three lines probably will still require an explicit cast, but I don't know how to test it locally.
Also combining all three lines makes the assignment too long to read, we will have to separate it to multiple lines which will have indentation instead of +=, I'm not sure if it's a good idea.